Fix missing inner enum import on operations

cortinico opened the pull request:
Previously inner types of complex types were not imported properly on
operations. This resulted in Retrofit interfaces missing some imports.

This commit introduces methods to inspect and import the inner types
used as result types of operations.

Fixes #76

Ping @macisamuele for the review

macisamuele commented on plugin/src/main/java/com/yelp/codegen/KotlinGenerator.kt:
```diff
@@ -142,10 +142,27 @@ class KotlinGenerator : SharedCodegen() {
+            .replace(" ", "")
+            .removeSuffix(">")
+            .removePrefix("List<")
+            .removePrefix("Array<")
```
As far as I can understand from the code the method will ever be called with `baseType` set to the type that will be output-ed on the file.
If the understanding is correct, then `Array<` should never be present

cortinico replied:
Yup, you're right. Let's refactor this out :)

macisamuele commented on plugin/src/main/java/com/yelp/codegen/KotlinGenerator.kt:
```diff
@@ -142,10 +142,27 @@ class KotlinGenerator : SharedCodegen() {
+    public override fun listTypeUnwrapper(baseType: String) = baseType
+            .replace(" ", "")
+            .removeSuffix(">")
+            .removePrefix("List<")
```
I'd rather prefer using [`kotlinTypeMapping["list"]`](https://github.com/Yelp/swagger-gradle-codegen/blob/master/plugin/src/main/java/com/yelp/codegen/utils/KotlinLangUtils.kt#L121) so it might be moved into `SharedCodegen` as it will be _kotlin_ indpendent

cortinico replied:
++

macisamuele commented on plugin/src/main/java/com/yelp/codegen/KotlinGenerator.kt:
```diff
@@ -142,10 +142,27 @@ class KotlinGenerator : SharedCodegen() {
+            .removePrefix("Array<")
+
+    public override fun isListTypeWrapped(baseType: String) =
+            baseType.endsWith('>') && (baseType.startsWith("Array<") || baseType.startsWith("List<"))
```
As before: `Array` should not be possible and I would replace List with `kotlinTypeMapping["list"]`

macisamuele commented on plugin/src/main/java/com/yelp/codegen/KotlinGenerator.kt:
```diff
@@ -142,10 +142,27 @@ class KotlinGenerator : SharedCodegen() {
+    public override fun mapTypeUnwrapper(baseType: String) = baseType
+            .replace(" ", "")
+            .removeSuffix(">")
+            .removePrefix("Map<${typeMapping["string"]},")
```
ditto: `s/Map/typeMapping["map"]/`

cortinico commented:
@macisamuele ready for another pass :) I've addressed all the comments.

> PS: personally I would have replaced isMapTypeWrapped with a regex matching of 
> the mapTypeWrapper (but not blocking)

You're more than welcome to add it once this gets merged :)
